add guidance to using underscores or dots for dbt model naming

mirnawong1 opened the pull request:
this pr adds explicit guidance on how to name dbt models. it adds :

- we recommend using underscores and not dots
- dots are supported, however come with complexity.
- adds guidance to models doc, style doc, and links to style doc. 

this stemmed from user confusion and [slack thread](https://dbt-labs.slack.com/archives/C012TTNN0JK/p1705660484585649?thread_ts=1704991223.136339&cid=C012TTNN0JK)

Commit:
add context

Commit:
add context

Commit:
add link to discourse

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

sophiad96 commented:
@mirnawong1 I'm a bit worried about saying "dots are supported but with complexity". I'm happy to hear other perspectives on this, but I'm not sure we should state that dbt supports dots in model names if we don't suggest it as a best practice. Also, the customer that mentioned this was confused by the very fact that we didn't state that we don't officially support dots in model names in the documentation.

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

mirnawong1 commented on website/docs/best-practices/how-we-style/1-how-we-style-our-dbt-models.md:
```diff
@@ -8,6 +8,10 @@ id: 1-how-we-style-our-dbt-models
+- Use underscores for naming dbt models; avoid dots.
+  - ✅  `models_without_dots`
+  - ❌ `models.with.dots`
+  - Underscores are more compatible and clear across platforms. Dots can cause quoting and case sensitivity complexities, especially in systems like Snowflake. Databases may also misinterpret dots as schema or database name separators. [dbt uses dots](https://discourse.getdbt.com/t/why-cant-model-names-contain-dots/422) in its `unique_id` generation to differentiate resource types, package names, ...
```
folding in sophia's feedback
```suggestion
  - Use underscores because they’re more compatible and clear across platforms. Dots can cause
 dbt Cloud parser, quoting, and case sensitivity issues, especially in systems like Snowflake. Databases may also misinterpret dots as schema or database name separators. [dbt uses dots](https://discourse.getdbt.com/t/why-cant-model-names-contain-dots/422) in its `unique_id` generation to differentiate resource types, package names, and resource names, which is another reason to avoid dots in model name. Read the [discussion](https://github.com/dbt-labs/dbt-core/issues/3246) for more details.
```

mirnawong1 commented on website/docs/docs/build/sql-models.md:
```diff
@@ -29,7 +29,10 @@ dbt's Python capabilities are an extension of its capabilities with SQL models.
 - Each `.sql` file contains one model / `select` statement
 - The model name is inherited from the filename.
-- Models can be nested in subdirectories within the `models` directory
+- Use underscores for model names for better compatibility and clarity (such as `model_without_dots`); avoid using dots to prevent potential issues with database interpretations (`model.with.dots`). 
```
```suggestion
- Use underscores for model names for better compatibility and clarity (such as `model_without_dots`); avoid using dots to prevent issues with the dbt Cloud parser and database interpretations (`model.with.dots`). 
```

mirnawong1 committed:
Update website/docs/docs/build/sql-models.md

mirnawong1 committed:
Update website/docs/best-practices/how-we-style/1-how-we-style-our-dbt-models.md

mirnawong1 commented on website/docs/best-practices/how-we-style/1-how-we-style-our-dbt-models.md:
```diff
@@ -8,6 +8,11 @@ id: 1-how-we-style-our-dbt-models
+  - ✅  `models_without_dots`
+  - ❌ `models.with.dots`
+  - Use underscores because they’re more compatible and clear across platforms. Dots can cause
+ dbt Cloud parser, quoting, and case sensitivity issues, especially in systems like Snowflake. Databases may also misinterpret dots as schema or database name separators. [dbt uses dots](https://discourse.getdbt.com/t/why-cant-model-names-contain-dots/422) in its `unique_id` generation to differentiate resource types, package names, and resource names, which is another reason to avoid dots in mod ...
```
hey @sophiad96 amazing feedback! made some tweaks, can you re review and let me know what you think!

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

mirnawong1 commented:
asked tech review [here](https://dbt-labs.slack.com/archives/C012TTNN0JK/p1705660484585649?thread_ts=1704991223.136339&cid=C012TTNN0JK)

sophiad96 reviewed (approved):
Looks great, thank you @mirnawong1 !!

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

joellabes commented on website/docs/docs/build/sql-models.md:
```diff
@@ -29,7 +29,10 @@ dbt's Python capabilities are an extension of its capabilities with SQL models.
 - Each `.sql` file contains one model / `select` statement
 - The model name is inherited from the filename.
-- Models can be nested in subdirectories within the `models` directory
+- Use underscores for model names for better compatibility and clarity (such as `model_without_dots`); avoid using dots to prevent issues with the dbt Cloud parser and database interpretations (`model.with.dots`). 
```
```suggestion
- We strongly recommend using underscores for model names, not dots. For example, use `models/my_model.sql` instead of `models/my.model.sql`.
```

mirnawong1 replied:
thank you! tweaked it and brought the example out of parenthesis.

joellabes commented on website/docs/best-practices/how-we-style/1-how-we-style-our-dbt-models.md:
```diff
@@ -8,6 +8,11 @@ id: 1-how-we-style-our-dbt-models
+- Use underscores for naming dbt models; avoid dots.
+  - ✅  `models_without_dots`
+  - ❌ `models.with.dots`
+  - Use underscores because they’re more compatible and clear across platforms. Dots can cause
+ dbt Cloud parser, quoting, and case sensitivity issues, especially in systems like Snowflake. Databases may also misinterpret dots as schema or database name separators. [dbt uses dots](https://discourse.getdbt.com/t/why-cant-model-names-contain-dots/422) in its `unique_id` generation to differentiate resource types, package names, and resource names, which is another reason to avoid dots in mod ...
```
```suggestion
  - Most data platforms use dots to separate `database.schema.object`, so using underscores instead of dots reduces your need for [quoting](/reference/resource-properties/quoting) as well as the risk of issues in certain parts of dbt Cloud. For more background, refer to [this GitHub issue](https://github.com/dbt-labs/dbt-core/issues/3246).
```

mirnawong1 replied:
thanks for tidying this up! i think it might help to start off with the guidance of 'use underscores for model names..' and then continue with the why. 

i folded both in, what do you think about this:


> - Use underscores for model names, not dots. Most data platforms use dots to separate `database.schema.object`, so using underscores instead of dots reduces your need for [quoting](/reference/resource-properties/quoting) as well as the risk of issues in certain parts of dbt Cloud. For more background, refer to [this GitHub issue](https://github.com/dbt-labs/dbt-core/issues/3246).

joellabes replied:
You already have 
>- Use underscores for naming dbt models; avoid dots.
>    - ✅  `models_without_dots`
>    - ❌ `models.with.dots`

So I didn't think it needed repeating a third time in the context section

mirnawong1 replied:
Oh i see, i saw it was red so I thought it was removed! Got it

joellabes replied:
![image](https://github.com/dbt-labs/docs.getdbt.com/assets/7335046/c21cd381-6472-449e-9dba-ca242033f60b)
For me the only red section is the context section, which is red because it was accidentally spilling onto two lines and I pulled it back to one

joellabes commented on website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md:
```diff
@@ -133,38 +133,32 @@ The dbt Cloud IDE makes it possible to [build and view](/docs/collaborate/build-
+</detailsToggle>
+
+<detailsToggle alt_header="Can I be a contributor to dbt Cloud?">
+Anyone can contribute to the dbt project. And whether it's a dbt package, a plugin, dbt-core, or this documentation site, contributing to the open-source code that supports the dbt ecosystem is a great way to level yourself up as a developer, and give back to the community. See <a href="https://docs.getdbt.com/community/resources/oss-expectations">Contributing</a> for details on what to expect wh ...
```
I know you are just cleaning up the wrapper around this, but since we're here... this doesn't answer the question at all. 

```suggestion
As a proprietary product, dbt Cloud's source code isn't available for community contributions. If you want to build something in the dbt ecosystem, we encourage you to review [this article](/community/contributing/contributing-coding) about contributing to a dbt package, a plugin, dbt-core, or this documentation site. Participation in open-source is a great way to level yourself up as a developer, and give back to the community.
```

mirnawong1 replied:
oh wow, thanks so much for clearing this up! this was definitely incorrect 🙏

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

mirnawong1 committed:
Update website/docs/docs/build/sql-models.md

Co-authored-by: Joel Labes <joel.labes@dbtlabs.com>

mirnawong1 committed:
Update website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md

Co-authored-by: Joel Labes <joel.labes@dbtlabs.com>

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

mirnawong1 committed:
Update 1-how-we-style-our-dbt-models.md

Co-authored-by: Joel Labes <joel.labes@dbtlabs.com>

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

mirnawong1 commented:
hey @joellabes ! thnks again for your. help and can you rereview please? folded in your comments 🙏

mirnawong1 committed:
Merge branch 'current' into mwong-model-names

mirnawong1 commented:
going to dismiss your review since i've addressed your feedback @joellabes - thank you again and will merge now!

https://docs-getdbt-90n3rdc79-dbt-labs.vercel.app/best-practices/how-we-style/1-how-we-style-our-dbt-models
